### PR TITLE
chat: fix issues with sent indicators, various issues with replies

### PR DESCRIPTION
**Pull request by patosullivan**
Fixes LAND-1167
Fixes LAND-1185
Fixes LAND-1184
Fixes LAND-1183

Also fixes a misnamed type for reply responses.

Edit: Now fixes LAND-1176 as well. I also decided to reuse the DiaryCommentField component for Heap comments since the use case is nearly identical (and we were abusing HeapTextInput for that purpose before).

### CODE REVIEW

**Comment by patosullivan** [comment]
I have to change my fix for displaying the pending status on chat messages

**Comment by patosullivan** [comment]
Ok, we should be good to go.

**Comment by patosullivan** [comment]
Accidentally tagged 1172, meant to tag LAND-1176. Still can't repro 1172.

File: ui/src/chat/ChatMessage/ChatMessage.tsx [hunk 1]
```diff
@@ -191,7 +191,7 @@ const ChatMessage = React.memo<
         author: window.our,
         sent: essay.sent,
       });
-      const isDelivered = msgStatus === 'delivered' || status === 'delivered';
+      const isDelivered = msgStatus === 'delivered' && status === 'delivered';
```
[the remark this answers is not in the document]
Comment (arthyn) [reply ui/src/chat/ChatMessage/ChatMessage.tsx:194]: ah ok that makes sense

**Review by arthyn — APPROVED** [review]
lgtm

**Pull request merged**